### ISSUE TITLE
Better way of showing indeterminate progress

### ISSUE THREAD

**Issue by o11c** [tqdm/tqdm#228]
Currently, I have a trick to handle input of unknown size: I set the `.total` to the next power of 2 above the soon-to-be-current value just before calling `.update`.

This guarantees that the fraction will repeatedly go from 50% up to 99%, which is actually a pretty decent UX. However, the tqdm-printed metadata includes my "total" as if it's fact, not merely a guess.

A better example is shown in visionmedia/node-progress#121 - though perhaps we should increase the size of the chunk logarithmically between some bounds (possibly altered by `__length_hint__`)?

**Comment by CrazyPython** [comment]
On a related note, we need a better statistics printer for infinite iterators - like when you want to track the speed of comment reading from reddit for your bot.

**Comment by lrq3000** [comment]
@o11c with #223 (if it gets merged), you will be able to show a looping animation if that's what you wanted. We could also make one standard in the core lib, but it would be only for UX niceness (because there's no additional information gathered from seeing a looping animation).

@CrazyPython What statistics are you missing?

**Comment by CrazyPython** [comment]
@lrq3000 for example, a dashed bar with the bar speeding up when it's faster would be nice.

**Comment by lrq3000** [comment]
Please see #223 and try the following:

```
import time
from tqdm import tcrange, tqdm_custom

bar_format_ind_loop = {
                    'symbols_indeterminate': {'ascii': list('/-\|'),
                                      'loop': True},
                      }

bar_format_ind = {
                    'symbols_indeterminate': {'ascii': ['----{,_,">', '----{,_,*>'],  # little mouse :)
                                      'loop': False},
                      }

with tqdm_custom(ascii=True, bar_format=bar_format_ind_loop) as tt1:
    for i in xrange(100):
        time.sleep(0.1)
        tt1.update()

with tqdm_custom(ascii=True, bar_format=bar_format_ind) as tt2:
    for i in xrange(100):
        time.sleep(0.1)
        tt2.update()
```

Does it fit your need? :)

**Comment by CrazyPython** [comment]
@lrq3000 This seems good. Some recipes for the braille thing and other cool little progress bars should be in the README or some configurations in a separate module would be nice.

**Comment by lrq3000** [comment]
@CrazyPython Yes we can either provide a few examples to demonstrate the API in the Readme, and provide more examples in a script in the `examples/` folder. We can potentially include any cool example we are given in such a script, the `examples` folder is perfectly fit for that I think :)

**Comment by CrazyPython** [comment]
@lrq3000 Yes, but maybe to keep with the tqdm simplicity principle, we should keep the examples importable. Let's open an issue on this, meanwhile, merge.
